### PR TITLE
ledger: kv keys allocation optimizations

### PR DESCRIPTION
**Pull request by icorderi**
## Summary

This addresses most of the comments left in #5488
Resolves https://github.com/algorand/go-algorand/issues/5538

## Test Plan

Existing tests.

### CODE REVIEW

File: ledger/store/trackerdb/generickv/schema.go [hunk 1]
```diff
@@ -23,149 +23,378 @@ import (
...
+	separator = '-'
+	// this is used as a value greather than the `separator` to get all the keys with a given prefix
+	endRangeSeparator = '.'
 )
 
 // return the big-endian binary encoding of a uint64
-func bigEndianUint64(v uint64) []byte {
-	ret := make([]byte, 8)
-	binary.BigEndian.PutUint64(ret, v)
+func bigEndianUint64(v uint64) [8]byte {
```
Comment (algorandskiy) [inline ledger/store/trackerdb/generickv/schema.go:55]: does it really helped? what is `go build -gcflags "-m=3" ./ledger/store/trackerdb/generickv 2>&1 | grep schema.go` output about heap alloc?

Comment (icorderi) [reply ledger/store/trackerdb/generickv/schema.go:55]: it says everything can be inlined and nothing escapes, except for 
```
ledger/store/trackerdb/generickv/schema.go:115:13: make([]byte, 0, prefixLength + separatorLength + len(kvKey)) escapes to heap
```

The kvKey arg is variable length, so it escapes it to the heap, i don't know how to force it to the stack if it's small enough.

**Comment by algorandskiy** [comment]
```
=== Failed
=== FAIL: ledger/store/trackerdb/testsuite TestDualEngines/online-accounts-delete (0.01s)
    onlineaccounts_kv_test.go:433: 
                Error Trace:    /opt/cibuild/project/ledger/store/trackerdb/testsuite/onlineaccounts_kv_test.go:433
                                                        /opt/cibuild/project/ledger/store/trackerdb/testsuite/utils_test.go:74
                Error:          Received unexpected error:
                                inconsistent results between store engines
                Test:           TestDualEngines/online-accounts-delete

=== FAIL: ledger/store/trackerdb/testsuite TestDualEngines (0.04s)
```

**Comment by jsgranados** [comment]
Resolves #5538

**Review by AlgoAxel — APPROVED** [review]

**Pull request merged**